src/main/java/hudson/plugins/jira/JiraIssueUpdater.java

```diff
@@ -27,25 +33,39 @@
  *
  * @author Kohsuke Kawaguchi
  */
-public class JiraIssueUpdater extends Recorder implements MatrixAggregatable {
+public class JiraIssueUpdater extends Recorder implements MatrixAggregatable, SimpleBuildStep {
 
     private UpdaterIssueSelector issueSelector;
-
+    private SCM scm;
+    private List<String> labels;
+    
     @DataBoundConstructor
-    public JiraIssueUpdater(UpdaterIssueSelector issueSelector) {
-        this.issueSelector = issueSelector;
-    }
-
-    @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-        // Don't do anything for individual matrix runs.
-        if (build instanceof MatrixRun) {
-            return true;
-        }
-
-        return Updater.perform(build, listener, getIssueSelector());
-    }
-
+	public JiraIssueUpdater(UpdaterIssueSelector issueSelector, SCM scm, List<String> labels) {
+		super();
+		this.issueSelector = issueSelector;
+		this.scm = scm;
+		if(labels != null)
+			this.labels = labels;
+		else
+			this.labels = Lists.newArrayList();
+	}
+	
+	@Override
+	public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws InterruptedException, IOException 
+	{
+		if(run instanceof AbstractBuild)
+		{
+			AbstractBuild<?,?> abstractBuild = (AbstractBuild<?,?>) run;
+			Updater updater = new Updater(abstractBuild.getParent().getScm(), labels);
+			updater.perform(run, listener, getIssueSelector());
```

Comment:
Original behavior: 

```java
// Don't do anything for individual matrix runs.
         if (build instanceof MatrixRun) {
             return true;
        }
```

Now the code handle matrix runs as other abstract builds